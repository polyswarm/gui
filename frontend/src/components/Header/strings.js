const strings = {
  title: "PolySwarm Nectar Relay"
}
export default strings;