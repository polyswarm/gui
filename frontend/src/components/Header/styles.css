.Header {
    background-color: #8500ff;
    border-bottom: #9d3bf8 2px solid;
    color: white;
    font-size: 24px;
    height: 48px;
    line-height: 48px;
    padding-left: 20px;
}