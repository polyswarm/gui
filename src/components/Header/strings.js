const strings = {
  logo: 'PolySwarm logo',
  mainnet: 'Main: ',
  nct: ' NCT ',
  sidenet: 'Side: ',
  eth: ' ETH',
};

export default strings;
