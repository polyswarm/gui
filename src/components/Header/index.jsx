import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import Button from '../Button';
import RequestSpinner from '../RequestSpinner';
import Dropdown from '../Dropdown';
// Component imports
import strings from './strings';
class Header extends Component {
  constructor(props) {
    super(props);
    this.onBountyClickHandler = this.onBountyClickHandler.bind(this);
    this.onOfferClickHandler = this.onOfferClickHandler.bind(this);
    this.onBack = this.onBack.bind(this);
  }

  render() {
    const { props: { title, back, requests, actions, address, nct, eth } } = this;
    let image_path = '../public/img/polyswarm-white.svg';
    if (back) {
      image_path = '../public/img/back-arrow.svg';
    }
    return (
      <header className='Header'>
        <div className='Header-Row'>
          <div className='Header-Title'>
            <img onClick={this.onBack}
              className='Header-Logo'
              src={image_path}
              alt={strings.logo}/>
            <h3>{title}</h3>
            <RequestSpinner requests={requests}/>
          </div>
          {!back && (
            <div className='Header-Actions'>
              {actions.slice(0, 2).map((action) => {
                return(
                  <Button key={action.title}
                    onClick={action.onClick}
                    header>
                    {action.title}
                  </Button>
                );
              })}
              {actions.slice(2).length > 0 && (
                <Dropdown light>
                  {actions.slice(2).map((action) => {
                    return(
                      <p key={action.title}
                        onClick={action.onClick}>
                        {action.title}
                      </p>
                    );
                  })}
                </Dropdown>
              )}
            </div>
          )}
        </div>
        <div className='Header-Address'>
          <p className='Header-Divided'>
            {address}
          </p>
          <p className='Header-Divided'>
            {`${strings.mainnet}${nct}${strings.nct}${eth}${strings.eth}`}
          </p>
          <p>
            {`${strings.sidenet}${nct}${strings.nct}`}
          </p>
        </div>
      </header>
    );
  }

  onBack() {
    const { props: { onBack } } = this;
    if(onBack) {
      onBack();
    } 
  }

  onBountyClickHandler() {
    const { props: { onClick } } = this;
    if(onClick) {
      onClick();
    }
  }

  onOfferClickHandler() {
    const { props: { onOfferClick } } = this;
    if(onOfferClick) {
      onOfferClick();
    }
  }
}

Header.propTypes = {
  title: PropTypes.string,
  back: PropTypes.bool,
  actions: PropTypes.array,
  address: PropTypes.string,
  nct: PropTypes.string,
  eth: PropTypes.string,
  onBack: PropTypes.func,
  requests: PropTypes.array,

};
export default Header;
