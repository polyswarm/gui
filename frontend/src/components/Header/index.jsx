// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Project imports
// Component imports
import './styles.css';
import strings from './strings';

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
        <header className="Header">
          <h2>{strings.title}</h2>
        </header>
    );
  }

}
Header.proptypes = {

};
export default Header;