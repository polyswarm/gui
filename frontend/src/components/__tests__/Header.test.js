import React from 'react';
import {render, shallow} from 'enzyme';
import {renderToJson, shallowToJson} from 'enzyme-to-json';
import Header from '../Header';

it('renders without crashing', () => {
  const wrapper = render(<Header />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('Uses title PolySwarm ', () => {
  const wrapper = shallow(<Header/>);
  expect(wrapper.text()).toEqual('PolySwarm Nectar Relay');
});