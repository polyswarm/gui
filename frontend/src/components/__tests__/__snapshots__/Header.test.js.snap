// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  class="Header"
>
  <h2>
    PolySwarm Nectar Relay
  </h2>
</header>
`;
